### PR TITLE
Add Eslint

### PR DESCRIPTION
**Pull request by chavlji**

### CODE REVIEW

**Review by f-jauregui — APPROVED** [review]
Awesome! 🙌

File: package.json [hunk 1]
```diff
@@ -26,7 +26,8 @@
     "storybook:start": "start-storybook --docs -p 6006",
     "storybook:build": "build-storybook --docs -o public --quiet",
     "styles:start": "webpack --watch --mode=development",
-    "styles:build": "webpack --mode=production"
+    "styles:build": "webpack --mode=production",
+    "lint": "eslint . --ext .js --ext .ts --ext .tsx"
```
Comment (mplesec) [inline package.json:30]: NIT: you can list the extensions like this: `--ext .js,.ts,.tsx`

File: package.json [hunk 2]
```diff
@@ -45,12 +46,17 @@
     "@storybook/theming": "~6.1.4",
     "@storybook/web-components": "~6.1.4",
     "@types/jest": "~26.0.15",
+    "@typescript-eslint/eslint-plugin": "^4.8.2",
```
Comment (mplesec) [inline package.json:49]: Why do these have `^`?

File: .eslintrc.js [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+    'no-plusplus': 0,
+    'no-param-reassign': 0,
+    'no-mixed-operators': 0,
+    'no-continue': 0,
+    'no-console': 0,
+    'no-prototype-builtins': 0,
+    'no-nested-ternary': 0,
+    'default-case': 0,
+    'space-before-function-paren': 0,
+    'guard-for-in': 0,
```
Comment (mplesec) [inline .eslintrc.js:20]: Would make sense to provide a description for why each exception.

File: .eslintrc.js [hunk 4]
```diff
@@ -0,0 +1,22 @@
+module.exports = {
+  "parserOptions": {
+    "project": "./tsconfig.json"
+  },
+  parser: '@typescript-eslint/parser',
+  plugins: ['@typescript-eslint'],
+  extends: ['airbnb-typescript/base'], // this is optional
```
Comment (mplesec) [inline .eslintrc.js:7]: You should also extend the recommended config (`"eslint:recommended"`).

Comment (mplesec) [reply .eslintrc.js:7]: And also `"plugin:@typescript-eslint/recommended"` since you're using the `@typescript-eslint` plugin.

**Pull request merged**